toolkits/jira/arcade_jira/tools/sprint_planning.py

```diff
@@ -0,0 +1,410 @@
+        "Required list of board names or IDs to retrieve sprints for. "
+        "Can contain mixed board identifiers - "
+        "both numeric IDs (e.g., '123') and board names (e.g., 'My Scrum Board'). "
+        "Example: ['123', 'My Board', '456']",
+    ],
+    sprints_per_board: Annotated[
+        int,
+        "The maximum number of sprints to return per board. Must be between 1 and 50 inclusive. "
+        "Defaults to 50 for comprehensive sprint data. "
+        "Sprints are returned with the latest ones first.",
+    ] = 50,
+    offset: Annotated[
+        int,
+        "The number of sprints to skip before starting to return results for each board. "
+        "Used for pagination when boards have many sprints. Applied to each board individually. "
+        "For example, offset=5 would skip the first 5 sprints on each board. "
+        "Must be 0 or greater. Defaults to 0 (start from the most recent sprints).",
+    ] = 0,
+    state: Annotated[
+        str | None,
+        "Filter sprints by their current state. Valid values are: "
+        "'future' (not started), 'active' (currently running), 'closed' (completed). "
+        "You can specify multiple states as a comma-separated list (e.g., 'active,future'). "
+        "If not provided, returns sprints in all states. Example: 'active' or 'active,future'",
+    ] = None,
+    start_date: Annotated[
+        str | None,
+        "Optional start date filter for sprints in YYYY-MM-DD format. "
+        "Filters sprints that overlap with or occur after this date. "
+        "Can be used together with end_date to create a date range. "
+        "Example: '2024-01-01'",
+    ] = None,
+    end_date: Annotated[
+        str | None,
+        "Optional end date filter for sprints in YYYY-MM-DD format. "
+        "Filters sprints that overlap with or occur before this date. "
+        "Can be used together with start_date to create a date range. "
+        "Example: '2024-03-31'",
+    ] = None,
+    specific_date: Annotated[
+        str | None,
+        "Optional specific date in YYYY-MM-DD format to get sprints that are active on that date. "
+        "Returns sprints where the specific date falls between their start and end dates. "
+        "Cannot be used together with start_date or end_date. Example: '2024-02-15'",
+    ] = None,
+) -> Annotated[
+    dict[str, Any],
+    "A comprehensive dictionary containing sprint data for the specified boards. "
+    "Includes a 'boards' list with board metadata, 'sprints_by_board' dictionary mapping "
+    "board IDs to their sprint data, and an 'errors' list for any boards that "
+    "don't support sprints or couldn't be processed. Sprints are sorted with latest first.",
+]:
+    """
+    List sprints for Jira boards by name or ID with advanced filtering and pagination.
+    Requires a list of board identifiers and supports date-based filtering.
+    Returns valid responses for sprint boards and error messages for non-sprint boards.
+    """
```

Comment:
Depending on how evals go, I'd suggest specifying the relationships between specific_date, start_date, and end_date since their relationship is complex.